agent/tcs/handler/handler.go

```diff
@@ -95,7 +95,7 @@ func startTelemetrySession(params TelemetrySessionParams, statsEngine stats.Engi
 
 func startSession(url string, region string, credentialProvider *credentials.Credentials, acceptInvalidCert bool, statsEngine stats.Engine, publishMetricsInterval time.Duration, deregisterInstanceEventStream *eventstream.EventStream) error {
 	client := tcsclient.New(url, region, credentialProvider, acceptInvalidCert, statsEngine, publishMetricsInterval)
-	defer client.Close()
+	defer client.Disconnect()
```

Comment:
As discussed offline: I think this is misplaced and you actually want this changed on line 113.